### PR TITLE
Dscs 2471

### PR DESCRIPTION
**Pull request by shuangzhou**
This request provides method to retrieve work records in session for reporting purpose.

### CODE REVIEW

**Comment by shuangzhou** [comment]
@m-r-c @ninhnguyen @adityaburra @sjacob @weijunnla @nlahuwgeddes, could you please review?

**Comment by nlabuilder** [comment]
SonarQube analysis reported 7 issues

1. MAJOR [AmberSession.java](https://github.com/nla/amberdb/blob/9aaa89b316b4c9bece59b5727c68901523489a45/src/main/java/amberdb/AmberSession.java): This file has 1,139 lines, which is greater than 1,000 authorized. Split it into smaller files. [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=squid%3AS00104)
1. MAJOR [AmberQuery.java#L350](https://github.com/nla/amberdb/blob/9aaa89b316b4c9bece59b5727c68901523489a45/src/main/java/amberdb/graph/AmberQuery.java#L350): This block of commented-out lines of code should be removed. [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. MAJOR [AmberQuery.java#L364](https://github.com/nla/amberdb/blob/9aaa89b316b4c9bece59b5727c68901523489a45/src/main/java/amberdb/graph/AmberQuery.java#L364): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=squid%3AS134)
1. MAJOR [AmberQuery.java#L376](https://github.com/nla/amberdb/blob/9aaa89b316b4c9bece59b5727c68901523489a45/src/main/java/amberdb/graph/AmberQuery.java#L376): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=squid%3AS134)
1. MINOR [AmberSession.java#L213](https://github.com/nla/amberdb/blob/9aaa89b316b4c9bece59b5727c68901523489a45/src/main/java/amberdb/AmberSession.java#L213): Document this method return value. [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [AmberSession.java#L334](https://github.com/nla/amberdb/blob/9aaa89b316b4c9bece59b5727c68901523489a45/src/main/java/amberdb/AmberSession.java#L334): Define a constant instead of duplicating this literal "collection" 3 times. [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=squid%3AS1192)
1. MINOR [AmberSession.java#L369](https://github.com/nla/amberdb/blob/9aaa89b316b4c9bece59b5727c68901523489a45/src/main/java/amberdb/AmberSession.java#L369): Define a constant instead of duplicating this literal "bibId" 3 times. [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=squid%3AS1192)

**Comment by nlabuilder** [comment]
Build finished.

**Comment by nlabuilder** [comment]
SonarQube analysis reported 4 issues

1. MAJOR [AmberSession.java](https://github.com/nla/amberdb/blob/91f735c469d170b77777ec4b180eb0697ad2e827/src/main/java/amberdb/AmberSession.java): This file has 1,139 lines, which is greater than 1,000 authorized. Split it into smaller files. [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=squid%3AS00104)
1. MINOR [AmberSession.java#L213](https://github.com/nla/amberdb/blob/91f735c469d170b77777ec4b180eb0697ad2e827/src/main/java/amberdb/AmberSession.java#L213): Document this method return value. [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [AmberSession.java#L334](https://github.com/nla/amberdb/blob/91f735c469d170b77777ec4b180eb0697ad2e827/src/main/java/amberdb/AmberSession.java#L334): Define a constant instead of duplicating this literal "collection" 3 times. [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=squid%3AS1192)
1. MINOR [AmberSession.java#L369](https://github.com/nla/amberdb/blob/91f735c469d170b77777ec4b180eb0697ad2e827/src/main/java/amberdb/AmberSession.java#L369): Define a constant instead of duplicating this literal "bibId" 3 times. [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=squid%3AS1192)

**Comment by nlabuilder** [comment]
Build finished.

File: src/main/java/amberdb/graph/AmberQuery.java [hunk 1]
```diff
@@ -284,60 +292,96 @@ private String generateBranchClause(QueryClause qc, int step) {
...
         
         // We don't handle branchList
-        QueryClause clause = clauses.get(0);
-        if (clause.branchList != null && clause.branchList.size() > 0) {
-            return null;
-        }
+        String labelList = "";
+        String headCol = "";
+        String tailCol = "";
+        if (clauses.size() == 1) {
```
Comment (scoen) [inline src/main/java/amberdb/graph/AmberQuery.java]: What if there are more than 1 clause?

**Pull request merged**